### ISSUE TITLE
Support for ReFT (Representational Finetuning)

### ISSUE THREAD

**Issue by creatorrr** [axolotl-ai-cloud/axolotl#1488]
### ⚠️ Please check that this feature request hasn't been suggested before.

- [X] I searched previous [Ideas in Discussions](https://github.com/OpenAccess-AI-Collective/axolotl/discussions/categories/ideas) didn't find any similar feature requests.
- [X] I searched previous [Issues](https://github.com/OpenAccess-AI-Collective/axolotl/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

ReFT seems very interesting. Dropping it here for further consideration. @winglian 

[https://huggingface.co/papers/2404.03592](Paper)
[https://github.com/stanfordnlp/pyreft](Library)

**Abstract**
> Parameter-efficient fine-tuning (PEFT) methods seek to adapt large models via updates to a small number of weights. However, much prior interpretability work has shown that representations encode rich semantic information, suggesting that editing representations might be a more powerful alternative. Here, we pursue this hypothesis by developing a family of Representation Finetuning (ReFT) methods. ReFT methods operate on a frozen base model and learn task-specific interventions on hidden representations. We define a strong instance of the ReFT family, Low-rank Linear Subspace ReFT (LoReFT). LoReFT is a drop-in replacement for existing PEFTs and learns interventions that are 10x-50x more parameter-efficient than prior state-of-the-art PEFTs. We showcase LoReFT on eight commonsense reasoning tasks, four arithmetic reasoning tasks, Alpaca-Eval v1.0, and GLUE. In all these evaluations, LoReFT delivers the best balance of efficiency and performance, and almost always outperforms state-of-the-art PEFTs. We release a generic ReFT training library publicly at https://github.com/stanfordnlp/pyreft.

### ✔️ Solution

---

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this feature has not been requested yet.
- [X] I have provided enough information for the maintainers to understand and evaluate this request.

**Comment by creatorrr** [comment]
On second thoughts, this probably doesn’t fit axoltl’s scope. To be specific, this feature request pertains to [LoReFT](https://github.com/frankaging/pyreft/tree/main/examples/loreft) but even that is very simple to achieve directly with their library and an axolotl integration is probably overkill. Closing for now.
